site/_wiki/Documentation/ConfigurationGuide.md

```diff
@@ -0,0 +1,418 @@
+  has. If you do decide to specify them as null, please make sure to include a
+  OpenTabletDriver diagnostics with the pull request, or include the lengths in
+  the commit description.
+
+### Picking the Parser {#parser}
+
+Picking the parser is the most important part of ensuring your tablet functions
+correctly. Picking the wrong parser can result in missing or (sometimes subtly)
+broken features.
+
+You must use a class which implements the `Parse(byte[])` function.
+
+How to write a parser from scratch is out of scope of this guide, but if you
+find that a parser you're using needs a code change to work, it is usually a
+better idea to make a new parser that builds off it instead.
+
+To see which functionality is supported by a parser, please see the code for
+the parsers return paths in its `Parse(byte[])` function. An example is
+provided below.
+
+#### `XP_PenReportParser` example
+
+At the time of this writing, [`XP_PenReportParser`] can return 5 different
+types of reports (not in order)
+
+[`XP_PenReportParser`]: https://github.com/OpenTabletDriver/OpenTabletDriver/blob/f14a184fef67a52637ab494497a151044bafd553/OpenTabletDriver.Configurations/Parsers/XP_Pen/XP_PenReportParser.cs#L7-L21
+
+##### `OutOfRangeReport`
+
+Fairly selfdescript, it doesn't actually report any specific values to the
+OS but rather 'resets' the pen out.
+
+##### `XP_PenAuxReport`:
+
+Reports `IAuxReport`, an aux-buttons-only report, ie. no position coordinates.
+
+##### `XP_PenTabletReport`
+
+Reports the following reports:
+
+- `ITabletReport`
+  - The pen's pressure
+  - Pen buttons
+  - It is an `IAbsolutePositionReport` that also reports the following:
+    - The 2-point vector of the pen
+- `ITiltReport`
+  - The 2-point vector of the pen's tilt
+- `IEraserReport`
+  - Whether the eraser is active or not
+
+##### `XP_PenTabletOverflowReport`
+
+An almost identical version of the above `XP_PenTabletReport` except with
+larger ranges for the position coordinate and the data structure adjusted
+accordingly.
+
+##### `TabletReport`
+
+A barebones tablet report that only reports an `ITabletReport`
+
+### Init strings (Feature/OutputInitReport, InitializationStrings)
+
+These are almost always necessary to use to bring the tablet out of HID mode
+and into a more preferred mode, e.g. vendor mode.
+
+Vendor mode is the mode used by the official drivers, and allows
+OpenTabletDriver to support all of the tablets features, assuming the parser
+supports it.
+
+Getting these can be achieved in 2 ways:
+
+#### Guessing
+
+Copying the appropriate init strings from a sibling configuration is usually
+adequate.
+
+#### USB Packet capture
+
+This is usually done with Wireshark (and winpcap). A guide on this is planned
+but not in progress. See [opentabletdriver.github.io#171] for more information.
+
+[opentabletdriver.github.io#171]: https://github.com/OpenTabletDriver/opentabletdriver.github.io/issues/171
+
+### Device strings
```

Comment:
You haven't really mentioned how to get these device strings? For something as simple as this it would be nice to avoid the guesswork on a process that can have quite a bit of guesswork if you don't know what you are doing?

This is a bit of a mess (because i'm repiping unsorted knowledge directly from my brain without thinking about it first), and applies to initialization strings too but...

Inits for huion/gaomon (most common to use strings):
- 200 is *usually* the only one required if its present (sometimes 123 is required for aux)
- if 200 isn't present, its usually 100, accompanied sometimes by 123 for auxiliary key support (users should test without 123 first)


I'm not sure what to put for device string matching? maybe pull some examples from the huion and gaomon configs as you have different patterns for those and explain how to give them the most compatibility (regex for any 6 numbers instead of specific numbers etc etc) while not being verbose that it could match tablets that it shouldn't.


I might try and write something up and suggest it directly, trying to condense all of my personal experience down into a small document is... hard.